doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,229 @@
+To be sure that this always holds, we run JBMC on the example, with a
+reasonable `unwind` value:
+```
+jbmc Example.class --function core.Example.isPrime --unwind 10
+```
+and unsurprisingly JBMC doesn't agree, and prints an assertion failure
+(truncated here for readability):
+```
+ [...doSomething:(I)V.assertion.1] line 7 assertion at file core/Example.java: FAILURE
+```
+
+Rerunning the analysis with the `--trace` option, we can see find out which
+input value JBMC found to trigger the violation:
+```
+INPUT arg0i: 15
+```
+The value chosen by JBMC is arbitrary, and could as well be 9 or
+2084569161.
+
+\subsection jbmc-manual-library Java Library support
+
+The examples above only involve primitive types. In order to analyse code
+involving Java Library classes, the core models (located in the same
+repository) need to be added to the classpath:
+```
+--cp <path_to_cbmc>/jbmc/src/java_bytecode/library/core-models.jar
+
+```
+The core models library simulate the behaviour of the Java Library and
+contains some commonly used classes, such as Object, Class, String, Enum.
+Some JBMC-specific utilities reside in the `org.cprover` package.
+
+Using the models, JBMC can reason with a variety of the String operations
+from `java.lang.String`, e.g.
+
+```java
+5    public static void doSomething() {
+6        StringBuilder sb = new StringBuilder("abcd");
+7        StringBuilder sb2 = sb.insert(2, "$");
+8        assert sb2.toString().startsWith("abc");
+9    }
+
+```
+The following command line (note that the current directory is also added to
+the classpath):
+```
+jbmc Example.class --function core.Example.doSomething --cp
+<path_to_cbmc>/jbmc/src/java_bytecode/library/core-models.jar:.
+```
+will flag this violation (truncated):
+```
+[java::core.Example.doSomething:()V.assertion.1] line 8 assertion: FAILURE
+```
+Again, the trace shows the contradicting string:
+```
+  dynamic_object2={ 'a', 'b', '$', 'c', 'd' }
+```
+
+\subsections jbmc-manual-exceptions Exceptions
+
+In its analysis, JBMC can take into account exceptions thrown by user code or
+library classes, as well as runtime exceptions such as NullPointerException.
+To have JBMC reliably notify about all uncaught exceptions, the
+`--throw-runtime-exceptions` can be used.
+
+In the following code, only with `--throw-runtime-exceptions` will JBMC
+signal that the assertion on line 11 can be violated. Without that option,
+line 11 will be considered as unreachable, and hence the assertion to be valid.
+```java
+5     public static int exceptionHandling(String str) {
+6         int retval = 0;
+7         try {
+8              retval = str.length();
+9         } catch (NullPointerException except) {
+10            retval = -1;
+11            assert false;
+12        }
+13        return retval;
+14    }
+```
+
+To remove all the assertions checking for uncaught exceptions, the
+`--disable-uncaught-exception-check` can be used in combination with
+`--throw-runtime-exceptions`.
```

Comment:
It might be worth pointing out that in the JVM `assert` actually throws an AssertionError on violation. Thus it can be handled like any other 'exception' (Throwable, to be precise). This behaviour can be enabled with `--throw-assertion-error`. An assertion violation is then reported as an uncaught exception.

Reply:
Added this